include/DataConsistencyGroup.h

```diff
@@ -3,107 +3,111 @@
 #pragma once
 
 #include "TransferElementAbstractor.h"
-#include "VersionNumber.h"
-#include <unordered_set>
 
 namespace ChimeraTK {
 
   /********************************************************************************************************************/
 
+  class BaseMatcher;
```

Comment:
Better place this and the two implementations into their own namespace, e.g. ChimeraTK::DataConsistencyGroupDetail. Also it should be `MatcherBase`, since it is the base class for the matchers, not a matcher for bases...